TheMET/TheMET/Services/FeaturedArtService.swift

```diff
@@ -136,10 +138,20 @@ class FeaturedArtService {
                     self?.isFeaturedArtLoading = false
                     return
                 }
-                self?.storeFeaturedArtData(artId: randomId, date: Date.now)
-                self?.artFileManager?.write(art: object)
-                self?.featuredArt = .loaded(object)
-                self?.isFeaturedArtLoading = false
+                guard let imageURL = object.primaryImage else {
+                    self?.forceUpdateFeaturedArt()
+                    return
+                }
+                self?.imageLoader.loadImage(urlString: imageURL, completion: { image in
+                    guard image != nil else {
+                        self?.forceUpdateFeaturedArt()
```

Comment:
`forceUpdateFeaturedArt` would cause the whole cycle to be restarted including re-downloading the list of objects' identifiers. It doesn't make sense.
Instead you should remove the checked object-identifier from the already received list, pick another random id and try it.